src/app/components/registration/registration.component.ts

```diff
@@ -7,10 +9,55 @@ import { Component, OnInit } from '@angular/core';
 })
 export class RegistrationComponent implements OnInit {
 
-  constructor() { 
+  constructor(private storage: StorageService) { 
   }
 
+  registrationFrom = new FormGroup({
+    name: new FormControl<string>('', Validators.required),
+    phone: new FormControl<string>('', [Validators.required, Validators.pattern('^\\+380[0-9]{9}$')]),
+    email: new FormControl<string>('', [Validators.required, Validators.email]),
+    birthday: new FormControl<string>('', Validators.required),
+    password: new FormControl<string>('', Validators.required),
+    subscribtion: new FormControl<boolean>(false)
+  })
+
+  isRegistered: boolean;
+  registrationErrorMessege: string | null;
+
   ngOnInit(): void {
   }
 
+  isUniquePhone(phone: string): boolean {
+    if (this.storage.users.find(user => user.phoneNumber === phone)) {
```

Comment:
Pls, minimize it as `return !this.storage.users.find(user => user.phoneNumber === phone)`